### PR TITLE
[Diffuser] Include registered buffers in param_dict during FSDP weight loading

### PR DESCRIPTION
**Pull request by ShanningZhuang**
## Motivation

Fixes #19413

In `load_model_from_full_model_state_dict()`, line 247 builds `meta_sd` from
`model.state_dict()`, which includes both parameters and persistent buffers.
But line 248 builds `param_dict` from `model.named_parameters()` only, which
excludes buffers.

When the loading loop iterates over checkpoint keys and a key corresponds to
a registered buffer:
1. `meta_sd.get(name)` finds it (line 268) — OK
2. `param_dict.get(name)` returns `None` (line 290) — buffer is missing
3. `weight_loader` is always `None` for buffers — their custom loading logic
   is silently skipped

Note: `state_dict()` cannot be used in place of `named_parameters()` +
`named_buffers()` because `state_dict()` returns copies of tensors, losing
custom attributes like `weight_loader`. `named_parameters()` and
`named_buffers()` return the original objects where `weight_loader` is
preserved.

## Modifications

Add `param_dict.update(dict(model.named_buffers()))` after line 248,
consistent with how line 147 in the same file already handles both:

    for n, p in chain(model.named_parameters(), model.named_buffers()):

## Checklist
- [x] Format: `pre-commit run --all-files` passed
- [x] The fix is additive — existing behavior for parameters is unchanged
- [x] For buffers without `weight_loader`: `getattr` returns `None`, falls
      through to the same raw-tensor path as before
- [x] For buffers with `weight_loader`: now correctly found and used

## Review Process

1. Ping Merge Oncalls to start the PR flow. See the [PR Merge Process](https://github.com/sgl-project/sglang/blob/main/.github/MAINTAINER.md#pull-request-merge-process).
2. Get approvals from [CODEOWNERS](https://github.com/sgl-project/sglang/blob/main/.github/CODEOWNERS) and other reviewers.
3. Trigger CI tests with [comments](https://docs.sglang.io/developer_guide/contribution_guide.html#how-to-trigger-ci-tests) or contact authorized users to do so.
   - `/tag-run-ci-label`, `/rerun-failed-ci`, `/tag-and-rerun-ci`
4. After green CI and required approvals, ask Merge Oncalls to merge.

### CODE REVIEW

**Comment by ShanningZhuang** [comment]
uv run sglang generate \
    --model-path Wan-AI/Wan2.1-T2V-1.3B-Diffusers \
    --use-fsdp-inference \
    --dit-layerwise-offload false \
    --num-gpus 2 \
    --prompt "A curious raccoon" \
    --save-output
    I've run this code to ensure it can work. Noticed that we need to set dit-layerwise-offload false to use FSDP

File: python/sglang/multimodal_gen/runtime/loader/fsdp_load.py
Comment (yyy1000) [inline python/sglang/multimodal_gen/runtime/loader/fsdp_load.py]: I'm thinking your change might not work as no buffer has `weight_loader` attribute set on it. How can you check this PR could work? IMO the command is not enough.

Comment (ShanningZhuang) [reply python/sglang/multimodal_gen/runtime/loader/fsdp_load.py]: Yes. Currently no buffer use weight_loader, so it is a preventive fix. But I think it's still worth merging for two reasons:
1. Consistency within the same file — line 147 already uses chain(model.named_parameters(),model.named_buffers()).  
https://github.com/sgl-project/sglang/blob/2e053d6eb/python/sglang/multimodal_gen/runtime/loader/fsdp_load.py#L147
So it is better if line 248 should match.
2. Someone might load a custom diffusion model with registered buffers. If their model attaches weight_loader to a buffer, it would silently fail. However I didn't find the public model which use the buffer so it cannot be reproduce only by the command.
That said, if you think this isn't worth the merge, I'm happy to hear your thoughts. It's a 1-line
change so the risk is minimal either way.

Comment (ShanningZhuang) [reply python/sglang/multimodal_gen/runtime/loader/fsdp_load.py]: So the author of this issue https://github.com/sgl-project/sglang/issues/19413 mention that he is using a custom model which has weight_loader in his code and it failed. I think this is a preventive fix. And because it is simple 1 line modification it won't cause any other problem. It would be appreciated if there is some feedback, thks!
